### PR TITLE
PreservedObject has_one CompleteMoab not has_many

### LINKED ISSUE #1952: change PreservedObject to has_one relationship with CompleteMoab
It's currently set to `has_many` from previous notions of multiple-moab.

Part of #1891

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made? 🤔

part of #1891 - removing multiple moab code from prescat

closes #1952

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test create_preassembly_image_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/create_preassembly_image_spec.rb) on stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

### CODE REVIEW

File: spec/services/complete_moab_handler_update_version_spec.rb [hunk 1]
```diff
@@ -30,9 +30,9 @@
 
     context 'in Catalog' do
       before do
-        v2 = create(:preserved_object, druid: druid, current_version: 2, preservation_policy: default_prez_policy)
-        v2.complete_moabs.create!(
-          version: v2.current_version,
+        create(:preserved_object, druid: druid, current_version: 2, preservation_policy: default_prez_policy)
+        po.create_complete_moab!(
+          version: po.current_version,
```
Comment (ndushay) [inline spec/services/complete_moab_handler_update_version_spec.rb:35]: this reverts closer to what it was before PR #1946

**Review by ndushay — COMMENTED** [review]
@jmartin-sul  I am least confident of changes to the `db/README.md` file.

File: spec/lib/audit/moab_to_catalog_spec.rb [hunk 2]
```diff
@@ -191,7 +191,8 @@
 
       storage_dir_a_seed_result_lists = described_class.seed_catalog_for_dir(storage_dir_a)
       expect(storage_dir_a_seed_result_lists.count).to eq 1
-      expected_result_msg = 'db update failed: #<ActiveRecord::RecordInvalid: Validation failed: Preserved object has already been taken>'
+      expected_result_msg = 'db update failed: #<ActiveRecord::RecordNotSaved: Failed to remove the existing associated complete_moab. ' \
+                            'The record failed to save after its foreign key was set to nil.>'
```
Comment (ndushay) [inline spec/lib/audit/moab_to_catalog_spec.rb:195]: @jmartin-sul this is the new spec from your PR #1954;  I assume this tweak is ok with you.

File: db/README.md [hunk 3]
```diff
@@ -369,6 +382,7 @@ FROM "preserved_objects"
 GROUP BY "preserved_objects"."druid", "preserved_objects"."current_version", "endpoint_name"
 HAVING (count(zipped_moab_versions.id) = 0)
 ```
+
 </details>
 
 ```ruby
```
[the remark this answers is not in the document]
Comment (ndushay) [reply db/README.md:388]: I couldn't find such a place, or it was already fixed.  The closest were around lines 260 and 402, and it didn't seem to apply to either of those.

File: spec/services/complete_moab_handler_check_exist_spec.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@
   let(:incoming_size) { 9876 }
   let(:po) { PreservedObject.find_by!(druid: druid) }
   let(:ms_root) { MoabStorageRoot.find_by!(storage_location: 'spec/fixtures/storage_root01/sdr2objects') }
-  let(:cm) { po.complete_moabs.find_by!(moab_storage_root: ms_root) }
+  let(:cm) { CompleteMoab.find_by!(moab_storage_root: ms_root) }
```
[the remark this answers is not in the document]
Comment (ndushay) [reply spec/services/complete_moab_handler_check_exist_spec.rb:13]: there is no "druid" in complete moab_handler, so this has to be by storage root.

**Review by ndushay — COMMENTED** [review]
@jmartin-sul I think I addressed all your concerns except for the spec change ... which can't be done.

**Review by mjgiarlo — APPROVED** [review]

**Review by jmartin-sul — APPROVED** [review]

**Pull request merged**